apps/client/src/app/core/data-reporting/submarine-exploration-result-reporter.ts

```diff
@@ -0,0 +1,131 @@
+import { DataReporter } from './data-reporter';
+import { Observable } from 'rxjs/Observable';
+import { ofPacketType } from '../rxjs/of-packet-type';
+import { filter, map, shareReplay, startWith, tap, withLatestFrom } from 'rxjs/operators';
+import { merge } from 'rxjs';
+import { LazyDataService } from '../data/lazy-data.service';
+import { ExplorationResultReporter, ExplorationType } from './exploration-result.reporter';
+
+export class SubmarineExplorationResultReporter implements DataReporter, ExplorationResultReporter {
+  private statusList$;
+  private resultLog$;
```

Comment:
Could be nice having proper typings for these, they're implicit any and it makes the code harder to read and maintain.